util/containerutil/shell_shared.go

```diff
@@ -285,24 +286,38 @@ func (cco *commmandContextOutput) string() string {
 
 func (sf *shellFrontend) commandContextStrings(args ...string) (string, []string) {
 	allArgs := append(sf.globalCompatibilityArgs, args...)
-
 	return sf.binaryName, allArgs
 }
 
+func (sf *shellFrontend) commandContextOutputWithRetry(ctx context.Context, retries int, timeout time.Duration, args ...string) (*commmandContextOutput, error) {
+	var err error
+	for i := 0; i < retries; i++ {
+		timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
+		defer cancel()
+		output, cmdErr := sf.commandContextOutput(timeoutCtx, args...)
+		if cmdErr == nil {
+			return output, nil
+		}
+		err = cmdErr
+		if i < retries-1 {
+			sf.Console.VerbosePrintf("Command '%s' failed. Retrying...\n", strings.Join(args, " "))
```

Comment:
Small nit:
This isn't technically the full command we're running. This is missing the binary and the `globalCompatibilityArgs` args we might append.

It would be nice for this log to be accurate, but that might bloat the scope of this PR.

Reply:
Always displaying the most recent failure message here would be useful:

```suggestion
			sf.Console.Printf("Command '%s' failed: %s. Retrying...\n", strings.Join(args, " "), cmdErr)
```

If it's hidden behind a verbose, and the command initially fails, some users might now have earthly wait up to 9*10=90 seconds before the 10th occurrence works. The user would have no idea why earthly is running so slowly.